test/assert_checkcircuit.go

```diff
@@ -78,6 +78,9 @@ func (assert *Assert) CheckCircuit(circuit frontend.Circuit, opts ...TestingOpti
 					// 2- if we are not running the full prover;
 					// we need to run the solver on the constraint system only
 					if !opt.checkProver {
+						if opt.skipTestEngine {
```

Comment:
Why we're skipping the solving when `NoTestEngine` set? Usually I'm skipping the test engine when I have modified the solver(e.g. for testing hint override option. This doesn't work with test engine as test engine calls hints directly and cannot be overidden), but now I cannot call solver on its own at all?

Reply:
As an example, I have used it here: https://github.com/Consensys/gnark/blob/master/internal/regression_tests/issue_836_test.go#L92

Reply:
Yeah issue on this one... is that for GKR for now some circuits are "unsolvable" by the solver, totally a misused of the flag here I removed it. 

i.e. with GKR the hint is valid only with a backend, need to replace the placeholder hint with something valid for the solver.

Reply:
Indeed, when we run with test engine then we need to have a working hint. Or we could right now shortcut it by having a specific error type which essentially says "I'm a hint only working with actual backend - skip solving with test engine" etc.

Reply:
Working on right now on GKR - actually the GKR hints are not dependent on the backend. We can overload the hints already in the solver. This would fix this issue.

Should I do a separate PR for that or add to this one?